Dockerfile

```diff
@@ -0,0 +1,103 @@
+FROM alpine:latest
+
+# ENV CLJ_TOOLS_VERSION=1.9.0.348
+ENV LEIN_VERSION=2.8.1
+ENV LEIN_INSTALL=/usr/local/bin/
+ENV LEIN_GPG_KEY=2B72BF956E23DE5E830D50F6002AF007D1A7CC18
+ENV MAVEN_VERSION=3.5.2
+ENV MAVEN_HOME=/usr/lib/mvn
+
+WORKDIR /tmp
+
+RUN apk add --verbose --update --upgrade --no-cache \
+	bash \
+	build-base \
+	ca-certificates \
+	curl \
+	docker \
+	fontconfig \
+	git \
+	gnupg \
+	jq \
+	nodejs-npm \
+	openjdk8-jre \
+	openssh \
+	openssl \
+	postgresql \
+	py-pip \
+	python \
+	rsync \
+	ruby \
+	ruby-bundler \
+	ruby-irb \
+	ruby-rdoc \
+	tar \
+	the_silver_searcher \
+	wget
+
+#--- Maven (from https://github.com/Zenika/alpine-maven/blob/master/jdk8/Dockerfile) 
+ENV PATH=$PATH:$MAVEN_HOME/bin
+
+RUN wget http://archive.apache.org/dist/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz && \
+  tar -zxvf apache-maven-$MAVEN_VERSION-bin.tar.gz && \
+  rm apache-maven-$MAVEN_VERSION-bin.tar.gz && \
+  mv apache-maven-$MAVEN_VERSION /usr/lib/mvn
+
+# Todo: Uncomment when clj tools required (will need to install rlwrap)
+#--- Clojure-Tools
+# RUN curl -O https://download.clojure.org/install/linux-install-$CLJ_TOOLS_VERSION.sh \
+#  && chmod +x linux-install-$CLJ_TOOLS_VERSION.sh \
+#  ./linux-install-$CLJ_TOOLS_VERSION.sh
+
+#--- Leiningen (from  https://github.com/Quantisan/docker-clojure/blob/master/alpine/lein/Dockerfile)
+
+# Download the whole repo as an archive
+RUN mkdir -p $LEIN_INSTALL \
+  && wget -q https://raw.githubusercontent.com/technomancy/leiningen/$LEIN_VERSION/bin/lein-pkg \
+  && echo "Comparing lein-pkg checksum ..." \
+  && echo "019faa5f91a463bf9742c3634ee32fb3db8c47f0 *lein-pkg" | sha1sum -c - \
+  && mv lein-pkg $LEIN_INSTALL/lein \
+  && chmod 0755 $LEIN_INSTALL/lein \
+  && echo "Fetching lein standalone zip ..." \
+  && wget -q https://github.com/technomancy/leiningen/releases/download/$LEIN_VERSION/leiningen-$LEIN_VERSION-standalone.zip \
+  && wget -q https://github.com/technomancy/leiningen/releases/download/$LEIN_VERSION/leiningen-$LEIN_VERSION-standalone.zip.asc \
+  && (gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-key "$LEIN_GPG_KEY" || \
+      gpg --keyserver hkp://ipv4.pool.sks-keyservers.net --recv-key "$LEIN_GPG_KEY" || \
+      gpg --keyserver hkp://pgp.mit.edu:80 --recv-key "$LEIN_GPG_KEY") \
```

Comment:
gpg keyservers were failing occasionally with just one. This allows some redundancy when building an image